### ISSUE TITLE
Unable to find Element After Setting Context Back to Webview 

### ISSUE THREAD

**Issue by InvisibleExo** [appium/appium#12520]
## The problem

I'm having trouble with my code trying to find an Element By.name after setting context back to Webview.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.11.1
* Node.js version (unless using Appium.app|exe): 6.4.1
* Mobile platform/version under test: Android
* Real device or emulator/simulator: Emulator
* Automation Name: UIAutomator2

## Details

I've been working a search Element method that will attempt to find the Target Element based on Direction and how many swipe attempts the user requests and state whether it has found the Target or not. I've set up my method in a way where it can operate the swipes based on whatever is the default context. When testing out this method on a webpage, such as the Google Results page, I'm having a problem, where the Driver is unable to find an Element as simple as the Google Search Bar (name="q"), after Driver Sets Context back to Webview, (which was the default context.) 
The output states that the context is Webview (CHROMIUM), but it's unable to find Element using Selenium Locator Strategy By.name("q"). Page Source shows that the element exists by that name. Based on my research trying to figure out the cause of the issue, I'm not sure if the cause is coming calling context(), some setting with Chrome when it comes to contexts, or if the UIAutomator2 not given enough time to load the Webview Context. (Tried waiting for 15 seconds, but still was unable to find Element). I get the feeling this isn't an issue on Appium's side, but at this point, I'm stumped. Is there something I'm missing?

## Link to Appium logs

Server Logs: https://gist.github.com/InvisibleExo/5ff1c1ba617276b50b69b3a982a0a8d3

WebPage Source on When Finding Element: https://gist.github.com/InvisibleExo/25b25684292a72250acaa43c5fded9f0

## Code To Reproduce Issue [ Good To Have ]
```
`void swipeUntilFound(By method, int attempts, MobileElement targetElement, Direction dir) {
		int limitCount = 0;
		double startScrollPoint = 0;
		double endScrollPoint = 0;
		switch (dir) {
			case up:
				startScrollPoint = 0.35;
				endScrollPoint = 0.8;
				break;
			case down:
				startScrollPoint = 0.8;
				endScrollPoint = 0.35;
				break;
		default:
			System.out.println("This method only supports directions up and down");
			break;
		}
		
		List<MobileElement> searchAfterSwipe = driver.findElements(method);
		while (searchAfterSwipe.isEmpty() || limitCount < attempts ) {
			swipeVertical(startScrollPoint, endScrollPoint, 3000);
			System.out.println("Context now is: " + driver.getContext());
			searchAfterSwipe = driver.findElements(method);
			limitCount++;
		}
		if(limitCount == attempts) {
			System.out.println("Unable to find Specified Element.");
		}else {
			targetElement = driver.findElement(method);
			System.out.println("Element " + method.toString() + " was found.");
		}
	}`
```

Inside the swipeVertical method: 

```
void swipeVertical(double startPercentage, double finalPercentage, int duration) {
		driver.context(nativeContext);
		size = driver.manage().window().getSize();
		int width = (int) (size.width/2);
		int startPoint = (int) (size.getHeight() * startPercentage);
		int endPoint = (int) (size.getHeight() * finalPercentage);
		new AndroidTouchAction(driver)
			.press(PointOption.point(width, startPoint))
			.waitAction(WaitOptions.waitOptions(Duration.ofMillis(duration)))
			.moveTo(PointOption.point(width, endPoint))
			.release()
			.perform();
		driver.context(defaultContext);
	}
```

**Comment by mykola-mokhnach** [comment]
Please attach the final pageSource output when switching to web context. Also, I think it makes sense to enable touch actions for in web context (although it will only work for absolute coordinates)

**Comment by InvisibleExo** [comment]
@mykola-mokhnach, the link attached is called with the Logs Capture setup I made after the Test fails. WebPage Source on When Finding Element: https://gist.github.com/InvisibleExo/25b25684292a72250acaa43c5fded9f0

I made another run and made a Call to Capture Page Source if it gets in the Conditions that it's unable to find the Element: https://gist.github.com/InvisibleExo/9600ca24058fbfde2a7fdd481bebfe9b

**Comment by mykola-mokhnach** [comment]
In the recent appium@beta it is possible to execute touch actions without switching to web context (only actions with absolute coordinates will work). Please reinstall appium@beta and check if the stuff works without changing the context

**Comment by InvisibleExo** [comment]
@mykola-mokhnach, I'm able to execute Touch Actions without switching context. Using version: v1,13.0-beta.3 . I also figured out the issue was on my end. My code was able to find Element in webview context after switching from Native_App, but the While-loop kept running since the amount of attempts didn't reach its limit. 

I do like having the ability to perform Touch Actions without switching context. 

I now notice that sometimes some Android Emulators are unable to perform Touch Action swipes(or at  least move,) due to the specific device  shows a size of 0 x 0.  I created a gist Log for when one device or more gets this issue: https://gist.github.com/InvisibleExo/addc1056d3812e0c1857f2e7d483f00a
